PR: https://github.com/odoo/odoo/pull/214213

From: 9d24acbec8657c61fdba7d4774e463ddb863638d
From: thha-odoo
Date: 2025-07-08 02:14:54

Breaking data model changes scores: del:2 + add:0, change matches:
-    activities_overdue = fields.Integer(compute='_compute_activities')
-    activities_today = fields.Integer(compute='_compute_activities')

Total Changes: 32

[IMP] hr_recruitment: homepage (job position) UI/X

Making a cleaner recruitment job positions view:
* Adding activity and configure button directly in the view to avoid losing time clicking o_kanban_card_manage_settings -> Activities/Configuration
* Renaming
	- open slots -> new applications
	- open applications -> new applications
* Adding "in progress" (more convenient)
* Displaying company name when it's possible instead of the user_id (more convenient)

Cleaner recruitment job position for mobile:
* Displaying only usefull infos
* Displaying company name when it's possible instead of the user_id (more convenient)

On job position activities view:
* Adjusting spacing right to the assignee avatar

[IMP] #4762193

closes odoo/odoo#214213

Related: odoo/upgrade#7971
Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_recruitment/models/hr_job.py
+++ b/addons/hr_recruitment/models/hr_job.py
@@ -63,8 +63,7 @@ class HrJob(models.Model):
     currency_id = fields.Many2one("res.currency", related="company_id.currency_id", readonly=True)
     compensation = fields.Monetary(currency_field="currency_id")
 
-    activities_overdue = fields.Integer(compute='_compute_activities')
-    activities_today = fields.Integer(compute='_compute_activities')
+    activity_count = fields.Integer(compute='_compute_activities')
 
     job_properties = fields.Properties('Properties', definition='company_id.job_properties_definition')
 
@@ -97,19 +96,15 @@ class HrJob(models.Model):
         self.env.cr.execute("""
             SELECT
                 app.job_id,
-                COUNT(*) AS act_count,
-                CASE
-                    WHEN %(today)s::date - act.date_deadline::date = 0 THEN 'today'
-                    WHEN %(today)s::date - act.date_deadline::date > 0 THEN 'overdue'
-                END AS act_state
+                COUNT(*) AS act_count
              FROM mail_activity act
              JOIN hr_applicant app ON app.id = act.res_id
              JOIN hr_recruitment_stage sta ON app.stage_id = sta.id
             WHERE act.user_id = %(user_id)s AND act.res_model = 'hr.applicant'
-              AND act.date_deadline <= %(today)s::date AND app.active
+              AND app.active
               AND app.job_id IN %(job_ids)s
               AND sta.hired_stage IS NOT TRUE
-            GROUP BY app.job_id, act_state
+            GROUP BY app.job_id
         """, {
             'today': fields.Date.context_today(self),
             'user_id': self.env.uid,
@@ -118,10 +113,9 @@ class HrJob(models.Model):
         })
         job_activities = defaultdict(dict)
         for activity in self.env.cr.dictfetchall():
-            job_activities[activity['job_id']][activity['act_state']] = activity['act_count']
+            job_activities[activity['job_id']] = activity['act_count']
         for job in self:
-            job.activities_overdue = job_activities[job.id].get('overdue', 0)
-            job.activities_today = job_activities[job.id].get('today', 0)
+            job.activity_count = job_activities[job.id]
 
     @api.depends('application_ids.interviewer_ids')
     def _compute_extended_interviewer_ids(self):
@@ -360,27 +354,13 @@ class HrJob(models.Model):
         views = ['activity'] + [view for view in action['view_mode'].split(',') if view != 'activity']
         action['view_mode'] = ','.join(views)
         action['views'] = [(False, view) for view in views]
-        return action
-
-    def action_open_late_activities(self):
-        action = self.action_open_activities()
         action['context'] = {
             'default_job_id': self.id,
             'search_default_job_id': self.id,
-            'search_default_activities_overdue': True,
             'search_default_running_applicant_activities': True,
         }
         return action
 
-    def action_open_today_activities(self):
-        action = self.action_open_activities()
-        action['context'] = {
-            'default_job_id': self.id,
-            'search_default_job_id': self.id,
-            'search_default_activities_today': True,
-        }
-        return action
-
     @api.model
     def _action_load_recruitment_scenario(self):
 
